daffodil-runtime1/src/main/scala/edu/illinois/ncsa/daffodil/processors/parsers/Parser.scala

```diff
@@ -110,21 +109,66 @@ trait Parser
        * ProcessingError it throws a new SDE.
        * */
       case pe: ParseError => pstate.setFailed(pe)
+    } finally {
+      pstate.resetFormatInfoCaches()
     }
     if (pstate.dataProc.isDefined) pstate.dataProc.get.after(pstate, this)
     pstate.setMaybeProcessor(savedParser)
   }
 }
 
-// Deprecated and to be phased out. Use the trait Parser instead.
-abstract class ParserObject(override val context: RuntimeData)
+/**
+ * A PrimParser is a parser that contains no child parsers.
+ * Combinators are NOT PrimParser
+ *
+ * This trait is preferred over PrimParserObject.
+ */
+trait PrimParser
+  extends PrimProcessor
+  with Parser
+
```

Comment:
Scaladoc for this trait needed, and all the other traits and classes in this file.